tests/analysis/_rewriters/test_sympy_rewriter.py

```diff
@@ -83,3 +83,42 @@ def test_list_arguments_of_function(self, function, expected_args):
     def test_expand(self):
         expr = self.backend.as_expression("(a + b)*c + d*(log2(x) + 5)")
         assert self.rewriter(expr).expand() == expr.expand()
+
+    @pytest.mark.parametrize(
+        "method, expected_default_value",
+        [
+            ["free_symbols", set()],
+            ["expand", None],
+            ["simplify", None],
+            ["all_functions_and_arguments", set()],
+        ],
+    )
+    def test_default_return_values_when_expr_is_numeric(self, method, expected_default_value):
+        rewriter = self.rewriter(2)
+        if callable(x := getattr(rewriter, method)):
+            assert x() == expected_default_value or 2
+        else:
+            assert x == expected_default_value
+
+    @pytest.mark.parametrize(
+        "expression, symbol, assumption, simplified_expression, property_symbol_satisfies",
+        [
+            ("max(0, a)", "a", "a > 0", "a", "is_positive"),
+            ("min(0, a)", "a", "a < 0", "a", "is_negative"),
+        ],
+    )
+    def test_add_assumption_simplifies_basic_expressions(
+        self, expression, symbol, assumption, simplified_expression, property_symbol_satisfies
```

Comment:
[nitpick] I'd say that `property_symbol_satisfies` could be just `property` – easier to parse, no?